pkg/uprobetracer/tracer.go

```diff
@@ -57,14 +58,14 @@ const (
 	ProgUretprobe
 )
 
-type inodeUUID struct {
-	token uint64
-}
-
-func getInodeUUID(file *os.File) (inodeUUID, error) {
-	// TODO: use `kfilefields` to get a unique token for each inode,
-	// here we just use `fd` as token, making each file has a distinct token.
-	return inodeUUID{uint64(file.Fd())}, nil
+func getInodeUUID(file *os.File) (uint64, error) {
+	token, err := kfilefields.ReadFInodeFromFd(int(file.Fd()))
+	priv, err := kfilefields.ReadPrivateDataFromFd(int(file.Fd()))
```

Comment:
Since `getInodeUUID()` can be called several times for a single gadget, it would make sense to optimize `kfilefields.ReadFInodeFromFd` to create the kfilefield ebpf program only once. Maybe with:
```
func ReadFInodesFromsFd(fd []int) ([]uint64, error)
```

(this could be done in a future PR)